Lab7/README.md

```diff
@@ -1,20 +1,181 @@
 ![gablogo][gablogo]
 
-# Lab 7 - Security - KeyVault 
+# Lab 7 - Security - KeyVault
 
 ## Goal
 
-Add a KeyVault to the ARM tempalte. migrate the API and storage keys to the keyvault. Use the keyvault instead of the keys  in the ARM template and application.
+Add a KeyVault to the ARM tempalte. migrate the API and storage keys to the keyvault. Use the keyvault instead of the keys in the ARM template and application.
 
 ## Let's code!
 
-[Enter here a step by step tutorial]
+1. [Create an identity for the App](#1-Create-an-identity-for-the-App)
+1. [Create a KeyVault to store secrets](#2-Create-a-keyVault-to-store-secrets)
+1. [Give the app access to the KeyVault's secrets](#3-Give-the-app-access-to-the-KeyVault's-secrets)
+1. [Move storage account key to KeyVault](#4-Move-storage-account-key-to-KeyVault)
+1. [Modify the app to get the Storage Account Keys from KeyVault](#5-Modify-the-app-to-get-the-Storage-Account-Keys-from-KeyVault)
+1. [Reference](#Reference)
+
+## 1. Create an identity for the App
+
+In order to access different azure services our app will need an identity.
+
+1. Login to the [azure portal](https://portal.azure.com)
+2. Use the search bar to get to the "Active Directory" service
+
+![](medias/search-active-directory.png)
+
+3. On the left side select "App registrations" then click on "New application registration
+
+![](medias/select-app-registration.png)
+
+4. Select a name and assign a Sign-on URL (can be anything and doesn't need to point to a valid website)
+
+![](medias/create-form.png)
+
+5. Create a key for you app, click on settings, then keys, then add a new key and click save
+
+![](medias/create-key.png)
+
+6. Copy the Application ID, object Id and the newly created key value in a notepad, we'll need them later
+
+![](medias/app-values.png)
+
+## 2. Create a KeyVault to store secrets
+
+1. Open the ARM template we created in [Lab2](../Lab2/README.md)
+1. Add a new resource for KeyVault
+
+```json
+{
+    "type": "Microsoft.KeyVault/vaults",
+    "name": "[parameters('keyVaultName')]",
+    "apiVersion": "2016-10-01",
+    "location": "[resourceGroup().location]",
+    "properties": {
+        "accessPolicies": [],
+        "enabledForDiskEncryption": false,
+        "sku": {
+            "name": "standard",
+            "family": "A"
+        },
+        "enabledForTemplateDeployment": true,
+        "tenantId": "[subscription().tenantId]",
+        "enabledForDeployment": true,
+        "accessPolicies": []
+    },
+    "resources": [],
+    "dependsOn": []
+}
+```
+> TIP: note that we are using template expressions like `[resourceGroup().location]` and `[subscription().tenantId]` to fill out automatically some values relative to the location where the resource will be deployed.
```

Comment:
Excellent idea! Maybe we could add a reference to [Azure Resource Manager template functions](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions)